packages/client/src/task-queue-client.ts

```diff
@@ -0,0 +1,226 @@
+    const resp = await this.workflowService.getWorkerBuildIdCompatibility({
+      taskQueue,
+      namespace: this.options.namespace,
+    });
+    if (resp.majorVersionSets == null || resp.majorVersionSets.length === 0) {
+      return undefined;
+    }
+    return versionSetsFromProto(resp);
+  }
+
+  /**
+   * Fetches task reachability to determine whether a worker may be retired. The request may specify
+   * task queues to query for or let the server fetch all task queues mapped to the given build IDs.
+   *
+   * When requesting a large number of task queues or all task queues associated with the given
+   * build ids in a namespace, all task queues will be listed in the response but some of them may
+   * not contain reachability information due to a server enforced limit. When reaching the limit,
+   * task queues that reachability information could not be retrieved for will be marked with a
+   * `NotFetched` entry in {@link BuildIdReachability.taskQueueReachability}. The caller may issue
+   * another call to get the reachability for those task queues.
+   */
+  public async getBuildIdReachability(options: ReachabilityOptions): Promise<ReachabilityResponse> {
+    const resp = await this.workflowService.getWorkerTaskReachability({
+      namespace: this.options.namespace,
+      taskQueues: options.taskQueues,
+      buildIds: options.buildIds,
+      reachability: reachabilityTypeToProto(options.reachability),
+    });
+    return reachabilityResponseFromProto(resp);
+  }
+}
+
+/**
+ * Options for {@link TaskQueueClient.getBuildIdReachability}
+ */
+export type ReachabilityOptions = RequireAtLeastOne<BaseReachabilityOptions, 'buildIds' | 'taskQueues'>;
+
+/**
+ * There are different types of reachability:
+ *   - `NewWorkflows`: The Build Id might be used by new workflows
+ *   - `ExistingWorkflows` The Build Id might be used by open workflows and/or closed workflows.
+ *   - `OpenWorkflows` The Build Id might be used by open workflows
+ *   - `ClosedWorkflows` The Build Id might be used by closed workflows
+ */
+export type ReachabilityType = 'NewWorkflows' | 'ExistingWorkflows' | 'OpenWorkflows' | 'ClosedWorkflows';
+
+interface BaseReachabilityOptions {
+  /**
+   * A list of build ids to query the reachability of. Currently, at least one Build Id must be
+   * specified, but this restriction may be lifted in the future.
+   */
+  buildIds: string[];
+  /**
+   *  A list of task queues with Build Ids defined on them that the request is
+   *  concerned with.
+   */
+  taskQueues?: string[];
+  /** The kind of reachability this request is concerned with. */
+  reachability?: ReachabilityType;
+}
+
+export interface ReachabilityResponse {
+  /** Maps Build Ids to their reachability information. */
+  buildIdReachability: Map<string, BuildIdReachability>;
+}
+
+type ReachabilityTypeResponse = ReachabilityType | 'NotFetched';
```

Comment:
That rename doesn't make sense to me, it's not the response type - it's the type *in the response*. Keeping unspecified also seems sort of a loss since the whole point of the higher level API is to make things more clear, but unspecified is not clear.

Reply:
Still need to export this